packages/flutter/lib/src/material/navigation_drawer.dart

```diff
@@ -0,0 +1,416 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+import 'color_scheme.dart';
+import 'colors.dart';
+import 'drawer.dart';
+import 'ink_well.dart';
+import 'material.dart';
+import 'material_state.dart';
+import 'navigation_bar.dart';
+import 'navigation_drawer_theme.dart';
+import 'text_theme.dart';
+import 'theme.dart';
+
+/// Material 3 Navigation Drawer component.
+///
+/// Navigation drawers offer a persistent and convenient way to switch between
+/// primary destinations in an app.
+///
+/// The style for the icons and text are not affected by parent
+/// [DefaultTextStyle]s or [IconTheme]s but rather controlled by parameters or
+/// the [NavigationDrawerThemeData].
+///
+/// This widget holds a collection of destinations ([NavigationDrawerDestination]s) and
+/// customized widgets, e.g. customized headlines, section titles, dividers.
+///
```

Comment:
Perhaps something like:

```suggestion
/// The [children] are a list of widgets to be displayed in the drawer. These can be a
/// mixture of any widgets, but there is special handling for [NavigationDrawerDestination]s.
/// They are treated as a group and when one is selected, the [onDestinationSelected]
/// is called with the index into the group that corresponds to the selected destination.
///
```